### DISCUSSION TITLE
Add support for multi-thread emulation?

### DISCUSSION THREAD

**Discussion by Fubuki0x10DE** [86Box/86Box#3851] [Ideas]
I mean is emulating virtual CPU and virtual peripherals using multi host CPU cores/threads (for example 2 threads or 4 threads emulation, configurable).
On Windows Me/XP, playback WMV/WMA smoothly requires 300 MHz or faster CPU.
Host: Intel Core i5 6600 (4C 4T)
On Windows Me, Start - Programs - Accessories - Entertainment - Windows Millennium Edition Preview
![1](https://github.com/86Box/86Box/assets/141222574/9e8ba613-d151-4c25-af13-3ede71be1157)

**Comment by eddmanx** [comment]
It's already as multi-threaded as it could practically be.

**Comment by lazerode** [comment]
I have a 10nm 4C/8T Mobile Core i5-1035G1 (Ice Lake-U). Sunny Cove (the microarchitecture used by Ice Lake) has more IPC than Skylake (the microarchitecture usec by 6th to 10th gen desktop Core CPUs). 300 MHz is a bit stable with the NDR and when idle, runs at a speeed no high than 67%.

**Comment by jriwanek** [comment]
The way emulation works means that fully threading 86box would actually likely reduce performance, not increase it, and make the code significantly more complex too. We get asked this question quite often btw. It won't be happening, as, as I said, it wouldn't provide any actual benefit.

**Comment by Torinde** [comment]
Related:
- https://github.com/86Box/86Box/discussions/3612#discussioncomment-6809855
